clients/web/src/views/widgets/HierarchyWidget.js

```diff
@@ -117,6 +118,18 @@ var HierarchyWidget = View.extend({
 
         this._showActions = _.has(settings, 'showActions') ? settings.showActions : true;
         this._showItems = _.has(settings, 'showItems') ? settings.showItems : true;
+
+        /*
+         * require that selectFolder be set *and* that a folder select callback
+         * be provided
```

Comment:
Why require both settings to be present to enable this?  If there really has to be two options here, I think an or condition would be more appropriate (making the selection callback a noop).